src/entities/pools/gyro2/gyro2Pool.ts

```diff
@@ -247,8 +247,8 @@ export class Gyro2Pool implements BasePool {
         sqrtAlpha: bigint;
         sqrtBeta: bigint;
     } {
-        const tIn = this.tokenMap.get(tokenIn.wrapped);
-        const tOut = this.tokenMap.get(tokenOut.wrapped);
+        const tIn = this.tokenMap.get(tokenIn.address);
```

Comment:
Any reason to update this logic?

Reply:
I considered this an oversight from my side. I copied code from other pools and didn't change that.
I'm not really sure what this `wrapped` means, so I thought it would make more sense to get the token based on its `address`.
Should I revert it to use `wrapped` instead?

Reply:
`wrapped` allows the sor and pool side of things to use the same `Token` class - so that native ETH (and similar assets) can be passed around while any pool / math uses the `wrapped` balance. If a `Token` class is not initialized with a different `wrapped` address it defaults to the token address.

I'll think through if there's a cleaner way to do this now that repo structure is mostly in place

Reply:
Got it! I reverted to use wrapped instead of address for now. ✅